database/transaction.rs

```diff
@@ -185,10 +180,19 @@ impl<D: DurabilityClient> TransactionWrite<D> {
         };
         commit_profile.things_finalised();
         drop(self.type_manager);
-        match snapshot.commit(commit_profile) {
-            Ok(_) => (profile, Ok(())),
-            Err(err) => (profile, Err(DataCommitError::SnapshotError { typedb_source: err })),
-        }
+        (profile, Ok((self.database, snapshot)))
+    }
+
+    pub fn commit(mut self) -> (TransactionProfile, Result<(), DataCommitError>) {
```

Comment:
Make this commit method delegate to database's `data_commit`.
TODO: remove this method and update the test accordingly on a follow-up PR